File: server/gzang-common/src/main/java/com/gzang/app/entity/Budget.java
package com.gzang.app.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 预算实体类
 *
 * @author G-Zang Team
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_budget")
public class Budget extends BaseEntity {

    /**
     * 所属用户ID
     */
    @TableField("user_id")
    private Long userId;

    /**
     * 所属公司ID（个人预算为NULL）
     */
    @TableField("company_id")
    private Long companyId;

    /**
     * 账本ID
     */
    @TableField("book_id")
    private Long bookId;

    /**
     * 分类ID（NULL表示总预算）
     */
    @TableField("category_id")
    private Long categoryId;

    /**
     * 预算金额
     */
    @TableField("amount")
    private BigDecimal amount;

    /**
     * 已使用金额
     */
    @TableField("used_amount")
    private BigDecimal usedAmount;

    /**
     * 预算周期类型 (1:月预算, 2:年预算, 3:周预算)
     */
    @TableField("period_type")
    private Integer periodType;

    /**
     * 预算周期开始日期
     */
    @TableField("period_start")
    private LocalDateTime periodStart;

    /**
     * 预算周期结束日期
     */
    @TableField("period_end")
    private LocalDateTime periodEnd;

    /**
     * 预算名称
     */
    @TableField("name")
    private String name;

    /**
     * 预警阈值（百分比，如80表示80%时预警）
     */
    @TableField("warning_threshold")
    private Integer warningThreshold;

    /**
     * 是否启用预警
     */
    @TableField("warning_enabled")
    private Boolean warningEnabled;

    /**
     * 备注
     */
    @TableField("remark")
    private String remark;
}

File: apps/mobile/src/api/budget.ts
/**
 * 预算相关 API
 * 后端接口: /api/mobile/budgets
 */
import { api } from './index';

/** 预算响应对象 */
export interface Budget {
  id: number
  userId: number
  companyId?: number
  bookId?: number
  categoryId?: number
  categoryName?: string
  categoryIcon?: string
  categoryColor?: string
  amount: number
  usedAmount: number
  remainingAmount: number
  usageRate: number
  periodType: number // 1=月预算, 2=年预算, 3=周预算
  periodTypeName: string
  periodStart: string
  periodEnd: string
  name: string
  warningThreshold: number
  warningEnabled: boolean
  isWarning: boolean
  remark?: string
  createTime: string
  updateTime?: string
}

/** 创建预算请求 */
export interface CreateBudgetRequest {
  name: string
  bookId?: number
  categoryId?: number
  amount: number
  periodType: number
  periodStart?: string
  periodEnd?: string
  warningThreshold?: number
  warningEnabled?: boolean
  remark?: string
}

/** 更新预算请求 */
export interface UpdateBudgetRequest {
  name?: string
  amount?: number
  periodStart?: string
  periodEnd?: string
  warningThreshold?: number
  warningEnabled?: boolean
  remark?: string
}

/**
 * 获取预算列表
 */
export async function getBudgets(params?: {
  bookId?: number
  periodType?: number
}): Promise<Budget[]> {
  return api.get<Budget[]>('/budgets', params);
}

/**
 * 获取预算详情
 */
export async function getBudget(id: number): Promise<Budget> {
  return api.get<Budget>(`/budgets/${id}`);
}

/**
 * 创建预算
 */
export async function createBudget(data: CreateBudgetRequest): Promise<Budget> {
  return api.post<Budget>('/budgets', data);
}

/**
 * 更新预算
 */
export async function updateBudget(id: number, data: UpdateBudgetRequest): Promise<Budget> {
  return api.put<Budget>(`/budgets/${id}`, data);
}

/**
 * 删除预算
 */
export async function deleteBudget(id: number): Promise<void> {
  return api.delete(`/budgets/${id}`);
}

/**
 * 获取预警预算列表
 */
export async function getWarningBudgets(): Promise<Budget[]> {
  return api.get<Budget[]>('/budgets/warnings');
}

/**
 * 刷新预算使用金额
 */
export async function refreshBudgetUsedAmount(id: number): Promise<void> {
  return api.post(`/budgets/${id}/refresh-used`);
}

export const budgetApi = {
  getBudgets,
  getBudget,
  createBudget,
  updateBudget,
  deleteBudget,
  getWarningBudgets,
  refreshBudgetUsedAmount,
};

export default budgetApi;
